tests/test_snql_api.py

```diff
@@ -309,6 +309,29 @@ def test_record_queries(self, record_query_mock: Any) -> None:
             assert metadata["request"]["referrer"] == "test"
             assert len(metadata["query_list"]) == expected_query_count
 
+    @patch("snuba.settings.RECORD_QUERIES", True)
+    @patch("snuba.state.record_query")
+    def test_record_queries_on_error(self, record_query_mock: Any) -> None:
+        record_query_mock.reset_mock()
+        response = self.post(
+            "/events/snql",
+            data=json.dumps(
+                {
+                    "query": f"""MATCH (events)
+                        SELECT event_id, title, transaction, tags[a], tags[b], message, project_id
+                        WHERE timestamp >= toDateTime('{self.base_time.isoformat()}')
+                        AND timestamp < toDateTime('{self.next_time.isoformat()}')
+                        AND project_id IN tuple({self.project_id})
+                        AND tags[a] = 1
+                        LIMIT 5""",
```

Comment:
out of curiosity, why does this cause a 386?

Reply:
Because we are comparing a tag to an int. I couldn't think of a case that would consistently cause a Clickhouse error, so if/when we change this behaviour we'll have to change/remove the test.

Reply:
I rewrote the test to not depend on this behaviour. Should be better long term.